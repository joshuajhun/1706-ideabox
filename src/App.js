import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import Controls from './Controls';
import CardContainer from './CardContainer';

class App extends Component {
  constructor() {
    super()
    this.state = {
      ideas: []
    }
    this.addIdea = this.addIdea.bind(this)
    this.deleteIdea = this.deleteIdea.bind(this)
  }

  addIdea(idea) {
    const newState = [...this.state.ideas, idea]
    this.setState({ ideas: newState })
  }

  deleteIdea(id) {
    const newState = this.state.ideas.filter(idea => idea.id !== id)
    this.setState({ ideas: newState })
  }

  render() {
    return (
      <div className="App">
        <h1>IDEAS!</h1>
        <Controls addIdea={ this.addIdea } />
        <CardContainer ideas={ this.state.ideas } deleteIdea={ this.deleteIdea } />
      </div>
    );
  }
}

export default App;
