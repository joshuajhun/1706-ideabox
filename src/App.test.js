import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, mount } from 'enzyme'

describe('App', () => {
  it(' should render the controsl Component', () => {
    const renderedApp = shallow(<App />)

    expect(renderedApp.find('Controls').length).toEqual(1)
    expect(renderedApp.find('CardContainer').length).toEqual(1)

  });

  it('should have a default state', () => {
    const renderedApp = mount(<App />)
    const initialState = renderedApp.state().ideas

    expect(initialState).toEqual([])
  });

  it('should let us add to the app state', () => {
    const renderedApp = mount(<App />)
    const initialState = renderedApp.state().ideas
    const titleInput = renderedApp.find('.title-input')
    const descriptionInput = renderedApp.find('.description-input')
    const submitButton =  renderedApp.find('.submit-button')

    expect(initialState).toEqual([])

    titleInput.simulate('change', {target: { value: 'why tho' } })
    descriptionInput.simulate('change', {target: { value: 'programming is hard' } })

    submitButton.simulate('click')

    expect(renderedApp.state().ideas.length).toEqual(1)
  })
  


});
