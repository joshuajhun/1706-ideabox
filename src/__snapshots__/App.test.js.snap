// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App  should render the controsl Component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          IDEAS!
</h1>,
        <Controls
          addIdea={[Function]}
/>,
        <CardContainer
          deleteIdea={[Function]}
          ideas={Array []}
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "IDEAS!",
        },
        "ref": null,
        "rendered": "IDEAS!",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "addIdea": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "deleteIdea": [Function],
          "ideas": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            IDEAS!
</h1>,
          <Controls
            addIdea={[Function]}
/>,
          <CardContainer
            deleteIdea={[Function]}
            ideas={Array []}
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "IDEAS!",
          },
          "ref": null,
          "rendered": "IDEAS!",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "addIdea": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "deleteIdea": [Function],
            "ideas": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
